### PR TITLE
[FEMR-150] Don't let a user enter the same password that they currently have

### PR DESCRIPTION
**Pull request by btrvarun**
Don't let a user enter the same password that they currently have - solved

https://teamfemr.atlassian.net/browse/FEMR-150

### CODE REVIEW

File: app/femr/business/services/system/UserService.java [hunk 1]
```diff
@@ -299,6 +298,11 @@ private void encryptAndSetUserPassword(IUser user) {
         user.setPassword(encryptedPassword);
     }
 
+    public Boolean checkOldPassword(String newpassword, String oldpassword)
```
Comment (kevinzurek) [inline app/femr/business/services/system/UserService.java]: This needs the @override annotation to tell the compiler you are overriding a method in the IUserService interface (see other methods in the same class)

File: app/femr/business/services/core/IUserService.java [hunk 2]
```diff
@@ -116,4 +116,7 @@
      * and/or errors if they exist.
      */
     ServiceResponse<IUser> update(IUser currentUser, Boolean isNewPassword);
+
+    Boolean checkOldPassword(String newpassword,String oldpassword);
+    //Check if the old password is equal to the new password
```
Comment (kevinzurek) [inline app/femr/business/services/core/IUserService.java]: If you use the javadoc template (see the comment on the update() method above) then you can click Ctrl+Q in IntelliJ while your cursor is on the method name to see what the method does.

**Comment by btrvarun** [comment]
Added both corrections

**Comment by kevinzurek** [comment]
Awesome! - @kdunlap i think we're good here

**Comment by kdunlap** [comment]
This all looks good to me. Nice work @btrvarun!

**Comment by kdunlap** [comment]
Manually merged

**Pull request closed without merging**